File: go-code/inline_function/inline-function.go
package inline_function

type Customer struct {
	Name     string
	Location string
}

func reportLine(customer Customer) []string {
	lines := make([]string, 2)
	gatherCustomerData(lines, customer)

	return lines
}

func gatherCustomerData(lines []string, customer Customer) {
	lines = append(lines, customer.Name)
	lines = append(lines, customer.Location)
}

File: java-code/src/main/java/refactoring/inline_function/InlineFunction.java
package refactoring.inline_function;

import java.util.ArrayList;
import java.util.List;

public class InlineFunction {

  public List<String> reportLines(Customer customer) {
    var lines = new ArrayList<String>();

    gatherCustomerData(lines, customer);

    return lines;
  }

  private void gatherCustomerData(ArrayList<String> lines, Customer customer) {
    lines.add("name: " + customer.getName());
    lines.add("location: " + customer.getLocation());
  }

}
